tools/clang/unittests/HLSL/ValidationTest.cpp

```diff
@@ -3842,22 +3861,30 @@ TEST_F(ValidationTest, ValidatePrintfNotAllowed) {
 
 TEST_F(ValidationTest, ValidateVersionNotAllowed) {
   if (m_ver.SkipDxilVersion(1, 6)) return;
-  std::string maxValMinor = std::to_string(m_ver.m_ValMinor);
-  std::string higherValMinor = std::to_string(m_ver.m_ValMinor + 1);
-  std::string maxDxilMinor = std::to_string(m_ver.m_DxilMinor);
-  std::string higherDxilMinor = std::to_string(m_ver.m_DxilMinor + 1);
+  // When validator version is < dxil verrsion, compiler has a newer version
+  // than validator.  We are checking the validator, so only use the validator
+  // version.
+  // This will also assume that the versions are tied together.  This has always
+  // been the case, but it's not assumed that it has to be the case.  If the
+  // versions diverged, it would be impossible to tell what DXIL version a
+  // validator supports just from the version returned in the IDxcVersion
+  // interface, without separate knowledge of the supported dxil version based
+  // on the validator version.  If these versions must diverge in the future, we
+  // could rev the IDxcVersion interface to accomodate.
```

Comment:
Since I added this originally, it's my fault for getting it wrong!  I think this fixes it, and I tried to document my thoughts here for future reference, since this stuff can get tricky with mismatched compiler/validator.